### PR TITLE
Adjust TizenCM's interface and update readme

### PR DESCRIPTION
**Pull request by keyuan0701**

### CODE REVIEW

**Review by xujiangang-git — COMMENTED** [review]
checked

**Review by wb0716 — COMMENTED** [review]
Checked!

**Review by gkl54333 — COMMENTED** [review]
good

**Review by tianhao-srcn — COMMENTED** [review]
checked

**Review by xueli-hu — COMMENTED** [review]
good

File: README.md [hunk 1]
```diff
@@ -35,6 +35,18 @@ Please note that, In order to use this pacakge, `@tizentv/webIDE-common-tizentv`
...
+    * privilegeLevel: 'public'/'partner'
+    */
+    getTizenDistributorProfile(privilegeLevel);
+
+
+    usage:
+
+        let tizenCertMgr = new TizenCM('D:\resource');
+        await tizenCertMgr.init();
+        tizenCertMgr.createCert(certInfo.authorInfo.info);
```
Comment (prozanne) [inline README.md:47]: There is a mistypo at 67 line.
ceateAuthorCert => createAuthorCert

**Review by jianhua-git — COMMENTED** [review]
checked

**Review by fjk0401 — COMMENTED** [review]
Good  for me

File: lib/tizenCertificateManager.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ module.exports = class TizenCM {
         this.certPath = path.resolve(resourcePath, 'cert');
         this.authorPath = path.resolve(resourcePath, 'Author');
 
-        this.init();
+        //this.init();
```
Comment (chunhua-liu) [inline lib/tizenCertificateManager.js]: delete

Comment (keyuan0701) [reply lib/tizenCertificateManager.js]: done

File: README.md [hunk 3]
```diff
@@ -35,6 +35,18 @@ Please note that, In order to use this pacakge, `@tizentv/webIDE-common-tizentv`
     }
     */
     createCert(authorInfo);
+    getTizenDeveloperCA();
+    /*
+    * privilegeLevel: 'public'/'partner'
+    */
+    getTizenDistributorProfile(privilegeLevel);
+
+
```
Comment (chunhua-liu) [inline README.md]: delete supernumerary blank lines

File: README.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ Please note that, In order to use this pacakge, `@tizentv/webIDE-common-tizentv`
 
 ```js
     constructor(resourcePath) ;
-
+    init();
     /*
     authorInfo {
```
Comment (prozanne) [inline README.md:25]: In my opinion....
"author" is too redundant.

I think the name of previous parameters looks ok. What do you think?
const authorInfo = {
  keyfileName: '', 
  authorName: '',
  authorPassword: '',
  countryInfo: '',
  stateInfo: '',
  cityInfo: '',
  organizationInfo: '',
  departmentInfo: '',
  emailInfo: ''
}

File: lib/downloadManager.js [hunk 5]
```diff
@@ -1,7 +1,6 @@
 const fs = require('fs');
 const path = require('path');
 const readline = require('readline');
-const http = require('http');
 const request = require('request');
```
Comment (SejoongDeJang) [inline lib/downloadManager.js]: https://www.npmjs.com/package/request
This package was deprecated. Replace to another package.

File: README.md (hunk 3 above)
Comment (keyuan0701) [reply README.md]: done

File: lib/downloadManager.js [hunk 6]
```diff
@@ -157,28 +167,35 @@ async function unzipPkg(zipFileDir, unzipDir, filterStr) {
...
-		    console.info('[webide-common-tizentv]create dir:' + tmpDir);
-		    fs.mkdirSync(tmpDir);
-	  }
-
-	  let streamFile = fs.createWriteStream(certZipFile);
-	  console.info('[webide-common-tizentv]requet to download ' + certGenFoler);
-	  await request(certGenFoler).pipe(streamFile).on('close',async () => {
-		    console.info('[webide-common-tizentv]downloadSamsungCertfile(): download certificate file successful');
-		    //unzipSamsungCertifile();
+    let certGenFoler = 'https://developer.samsung.com/sdk-manager/repository/tizen-certificate-extension_2.0.42.zip';
```
Comment (SejoongDeJang) [inline lib/downloadManager.js]: Refer to your code, the `tizen-certificate-extesion` is disable to change version. Can you get a latest version of the `tizen-certificate-extension`?

Comment (SejoongDeJang) [reply lib/downloadManager.js]: And one more thing, what is a 'foler'? Is it a typo?

File: lib/downloadManager.js [hunk 7]
```diff
@@ -157,28 +167,35 @@ async function unzipPkg(zipFileDir, unzipDir, filterStr) {
...
+        for (let fileName of certAddfileArr) {
             if (fileName.indexOf(platform) > 0) {
                 certAddFileName = fileName;
             } else {
                 fs.unlinkSync(path.resolve(tmpDir, fileName));
             }
-    		}
+        }
         await unzipPkg(tmpDir, tmpDir, 'data/tools/certificate-manager/plugins');
         await unzipPkg(tmpDir, toolsDir + path.sep + 'SamsungCertificate', 'res/ca');
```
Comment (SejoongDeJang) [inline lib/downloadManager.js]: As I checked the final output, I checked followed 
``` bash
org.tizen.common.cert_1.0.0.201911210714\res\ca>ls
gear_test_author_CA.cer    gear_test_public_CA.cer          samsung_tizen_dev_public2.crt  vd_tizen_dev_platform2.crt
gear_test_partner_CA.cer   samsung_tizen_dev_partner2.crt   vd_tizen_dev_author_ca.cer     vd_tizen_dev_public2.crt
gear_test_platform_CA.cer  samsung_tizen_dev_platform2.crt  vd_tizen_dev_partner2.crt
```
And do you remove gear and VD files?

File: lib/tizenCertificateManager.js [hunk 8]
```diff
@@ -209,23 +205,23 @@ module.exports = class TizenCM {
...
-        else if (privilegeLevel == 'partner') {
+        let distributorCA = '';
+        let distributorCertPath = '';
+        if (privilegeLevel == 'partner') {
             distributorCA = distributorPartnerCA;
             distributorCertPath = distributorPartnerSigner;
         }
+        else {
+              distributorCA = distributorPublicCA;
+              distributorCertPath = distributorPublicSigner;
```
Comment (SejoongDeJang) [inline lib/tizenCertificateManager.js:221]:
``` js
return {
    distributorCA: distributorCA,
    distributorCertPath: distributorCertPath,
    distributorPassword: distributorPassword
};
```
Refer to return object, the `distributorPassword` is fixed to 'tizenpkcs12passfordsigner'. Is it right?

File: lib/downloadManager.js [hunk 9]
```diff
@@ -52,77 +52,87 @@ class PackageInfo {
...
 
-		let pkgName = 'temp.zip';
-		let pathArr = packagePath.split('/');
-		pkgName = pathArr[pathArr.length - 1];
+    let pkgName = 'temp.zip';
+    let pathArr = packagePath.split('/');
+    pkgName = pathArr[pathArr.length - 1];
+    let requestOpt = packagePath;
+
+    let userProxy = npmConf.get('http-proxy');
```
Comment (SejoongDeJang) [inline lib/downloadManager.js]: https://wesbos.com/let-vs-const Read this article. You may understand difference between `let` and `const`. Shortly, use `const` keyword if the variable is `object` or `array` type. You can change the property of `object` or `array`. 
``` js
const a = [1]
// a = [1]
a.push(2)
// a = [1, 2]
```
Immutable means block the assignment( `=` operator). And if you want to block change the property, than use `Object.freeze()`. https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

File: README.md (hunk 4 above)
Comment (keyuan0701) [reply README.md:25]: done

File: lib/tizenCertificateManager.js (hunk 8 above)
Comment (keyuan0701) [reply lib/tizenCertificateManager.js:221]: yes, it is fixed password for tizen distributor

File: lib/downloadManager.js (hunk 9 above)
Comment (keyuan0701) [reply lib/downloadManager.js]: done

File: lib/downloadManager.js (hunk 7 above)
Comment (keyuan0701) [reply lib/downloadManager.js]: These are downloaded cert files for samsung certification manager, vd files are necessary files. we just ensure samsung cert files are existed.

File: lib/downloadManager.js (hunk 6 above)
Comment (keyuan0701) [reply lib/downloadManager.js]: done

File: lib/downloadManager.js (hunk 5 above)
Comment (keyuan0701) [reply lib/downloadManager.js]: done

**Review by prozanne — APPROVED** [review]
Looks good to me

**Pull request merged**